### ISSUE TITLE
Confused by C++20 trailing requires-clause 

### ISSUE THREAD

**Issue by jwakely** [doxygen/doxygen#8726]
**Describe the bug**

A bogus warning:

```
Parsing file /tmp/x.cpp...
/tmp/x.cpp:27: warning: Found ';' while parsing initializer list! (doxygen could be confused by a macro call without semicolon)
```
This is caused by a C++20 requires-clause after the function parameter list:

```cpp
template<typename T> concept foo = true;

struct weak_ordering
{
  static constexpr int less = -1;
  static constexpr int equivalent = 0;
  static constexpr int greater = 1;
};

struct S
{
  template<typename T, typename Up>
    constexpr auto
    operator()(const T& t, const Up& u) const
    requires requires
    {
      { t < u } -> foo;
    }
    {
      if constexpr (false)
        return t <=> u;
      else
        {
          if (t < u)
            return weak_ordering::less;
          else if (u < t)
            return weak_ordering::greater;
        }
    }
};
```

**To Reproduce**
[requires-bug.tar.gz](https://github.com/doxygen/doxygen/files/7014732/requires-bug.tar.gz)

**Version**

1.9.3 (a71a76b622144877c248180487de4f4544032563)

**Comment by jwakely** [comment]
A slightly modified version of the same code adds a new warning:

```
Searching for member function documentation...
/tmp/x.cpp:30: warning: documented symbol 'else S::if' was not declared or defined.
```

```cpp
template<typename T> concept foo = true;

struct weak_ordering
{
  static constexpr int less = -1;
  static constexpr int equivalent = 0;
  static constexpr int greater = 1;
};

struct S
{
  template<typename T, typename Up>
    constexpr auto
    operator()(const T& t, const Up& u) const
    requires requires
    {
      { t < u } -> foo;
    }
    {
      if constexpr (false)
        return t <=> u;
      else
        {
          if (t < u)
            return weak_ordering::less;
          else if (u < t)
            return weak_ordering::greater;
          else
            return weak_ordering::equivalent;
        }
    }
};
```

[requires-bug-2.tar.gz](https://github.com/doxygen/doxygen/files/7014803/requires-bug-2.tar.gz)

**Comment by albert-github** [comment]
In the original example (i.e. the tar file) there is:
```
  static conster int equivalent = 0;
```
probably a typo that should read:
```
  static constexpr int equivalent = 0;
```
In both examples there is `requires requires` is this correct?

**Comment by jwakely** [comment]
Yes, sorry, I fixed the `conster` typo in the comment here but not the attached tar file. It doesn't affect the doxygen behaviour.

The `requires requires` is correct C++20 code.

**Comment by doxygen** [comment]
@jwakely Please verify if the referenced commit fixes the problem for you.

**Comment by jwakely** [comment]
Yes, that seems to work. Thanks!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
